src/AzSK.ADO/Framework/Core/STMapping/AzSKADOServiceMapping.ps1

```diff
@@ -325,23 +360,63 @@ class AzSKADOServiceMapping: CommandBase
                     if ($this.MappingType -eq "All" -or $this.MappingType -eq "VariableGroup") {
                         if(($varGrps | Measure-Object).Count -gt 0)
                         {
-                        $varGrps | ForEach-Object{
-                            $varGrpURL = ("https://{0}.visualstudio.com/{1}/_apis/distributedtask/variablegroups/{2}?api-version=6.1-preview.2") -f $this.OrgName, $this.projectId, $_;
-                            $varGrpObj = [WebRequestHelper]::InvokeGetWebRequest($varGrpURL);
-
-                            $releaseSTData = $this.ReleaseSTDetails.Data | Where-Object { ($_.releaseDefinitionID -eq $releaseObj[0].id) };
-                            if($releaseSTData){
-                                $variableGroupSTMapping.data += @([PSCustomObject] @{ variableGroupName = $varGrpObj.name; variableGroupID = $varGrpObj.id; serviceID = $releaseSTData.serviceID; projectName = $releaseSTData.projectName; projectID = $releaseSTData.projectID; orgName = $releaseSTData.orgName } )
+                            $apiURL = "https://{0}.visualstudio.com/_apis/Contribution/HierarchyQuery?api-version=5.0-preview.1" -f $this.OrgName;
+                            $sourcePageUrl = "https://{0}.visualstudio.com/{1}/_settings/adminservices" -f $this.OrgName, $this.ProjectName;
+
+                            $varGrps | ForEach-Object {
+
+                                try {
+                                    $varGrpURL = ("https://{0}.visualstudio.com/{1}/_apis/distributedtask/variablegroups/{2}?api-version=6.1-preview.2") -f $this.OrgName, $this.projectId, $_;                            
+                                    $varGrpObj = [WebRequestHelper]::InvokeGetWebRequest($varGrpURL);                              
+                                    if ($varGrpObj.Type -eq 'AzureKeyVault') {   
```

Comment:
if else condition are wrong.
As per this condition, if vargroup type is azure key vault then fetch value from ReleaseSTData, 
This is not correct we should find value in ReleaseSTData whatever type is variable group. 
In else condition(new subscription logic) we can check if vargroup type is azure key vault and ST mapping is not found in ReleaseSTData then proceed to get data using new subscription logic.
Same  for Build section.
Currently, it is doing if vargroup type is key vault then it is fetching data from ReleaseST/BuildSTData and for other type it is using new subscription logic, which is wrong.